p8e-util/src/main/kotlin/io/p8e/crypto/SmartKeySigner.kt

```diff
@@ -11,20 +12,47 @@ import com.fortanix.sdkms.v1.model.VerifyRequest
 import com.google.protobuf.Message
 import io.p8e.proto.Common.Signature
 import io.p8e.util.orThrow
-import io.p8e.util.toProtoUuidProv
+import io.p8e.util.toHex
+import io.p8e.util.toJavaPublicKey
+import org.bouncycastle.jcajce.provider.asymmetric.ec.BCECPublicKey
+import org.bouncycastle.pqc.jcajce.provider.BouncyCastlePQCProvider
 import java.lang.IllegalStateException
+import java.security.KeyFactory
+import java.security.PublicKey
+import java.security.interfaces.ECPublicKey
+import java.security.spec.X509EncodedKeySpec
+
+/**
+ * About SmartKey
+ *
+ * SmartKey™ powered by Fortanix is the world’s first cloud service secured with Intel® SGX. With SmartKey, you can
+ * securely generate, store, and use cryptographic keys and certificates, as well as other secrets such as passwords,
+ * API keys, tokens, or any blob of data. Your business-critical applications and containers can integrate with
+ * SmartKey using legacy cryptographic interfaces or using the native SmartKey RESTful interface.
+ *
+ * SmartKey uses built-in cryptography in Intel® Xeon® CPUs to help protect the customer’s keys and data from all
+ * external agents, reducing the system complexity greatly by removing reliance on characteristics of the physical
+ * boxes. Intel® SGX enclaves prevent access to customer’s keys or data by Equinix, Fortanix or any other cloud service
+ * provider.
+ *
+ * Unlike many hardware security technologies, Intel® SGX is designed to help protect arbitrary x86 program code.
+ * SmartKey uses Intel® SGX not only to help protect the keys and data but also all the application logic including
+ * role based access control, account set up, and password recovery. The result is significantly improved security
+ * for a key management service that offers the elasticity of modern cloud software and the hardware-based security
+ * of an HSM appliance, all while drastically reducing initial and ongoing costs.
+ *
+ * SmartKey is designed to enable businesses to serve key management needs for all their applications, whether they are
+ * operating in a public, private, or hybrid cloud.
+ */
 
 class SmartKeySigner(
-    appApiKey: String,
-    uuid: String
+    private val appApiKey: String,
+    private val keyUuid: String
 ): SignerImpl {
 
-    private var sObjUUID: String = uuid
-    private var apiKey: String = appApiKey
-
     init {
```

Comment:
This can be in a separate ticket, but given that this class now has many instances that could be signing simultaneously, I think we should make a bean for the actual api instance and have this class just hold a reference to the single client.

Likewise, if I remember correctly, the oauth token has some limited time duration so the singleton client class would need a background thread that updates the oauth token on a recurring basis.

Reply:
Good point, I'll create tickets for them.